### ISSUE TITLE
Provide a stable SONAME for shared library

### ISSUE THREAD

**Issue by tartina** [hydrogen-music/hydrogen#1293]
Hydrogen version *          : all
Operating system + version  : Linux all
Audio driver + version      : all

The SONAME of the shared library is bumped on every release, even if no incompatible changes have been made on the library. For better upgrading please provide a major SONAME, i.e. libhydrogen-core-1.so which does not change when API remains compatible and provide symbolic links to the real file, which can retain all minor versions, like other libraries do.

**Comment by cme** [comment]
Hi, is this causing any actual problems?

The library isn't currently expected to be used by anything other than the tools included with hydrogen itself. It's been changing a lot recently and isn't really a published interface, but if there's some other project that wants to make use of the Hydrogen core library, we can think about better supporting that.

**Comment by charbeljc** [comment]
> Hi, is this causing any actual problems?
> 
> The library isn't currently expected to be used by anything other than the tools included with hydrogen itself. It's been changing a lot recently and isn't really a published interface, but if there's some other project that wants to make use of the Hydrogen core library, we can think about better supporting that.

I'm currently making python bindings to libhydrogen-core, so I guess it could be useful for publishing my bindings and let It work with compatible installed hydrogen versions.

**Comment by tartina** [comment]
If this is a "private" library and nobody is supposed to use it, this is not a problem.
But as you see above, there is at least one external user of the library.
When you think the library is stable enough, I believe it could be a nice to have feature.

**Comment by theGreatWhiteShark** [comment]
> When you think the library is stable enough, I believe it could be a nice to have feature.

Yeah, that's the crucial requirement we do not meet yet. But as soon as the core approached a certain degree of stability, this would be indeed a nice feature.
